native/src/hooks/useLocation.ts

```diff
@@ -0,0 +1,141 @@
+import Geolocation, { GeolocationError, GeolocationResponse } from '@react-native-community/geolocation'
+import { useCallback, useEffect, useState } from 'react'
+import { openSettings, RESULTS } from 'react-native-permissions'
+import SystemSetting from 'react-native-system-setting'
+
+import { checkLocationPermission, requestLocationPermission } from '../utils/LocationPermissionManager'
+
+type SuccessfulLocationState = { status: 'ready' }
+type UnavailableLocationState = {
+  status: 'unavailable'
+  message: 'noPermission' | 'notAvailable' | 'timeout' | 'loading'
+}
+
+type LocationStateType = SuccessfulLocationState | UnavailableLocationState
+
+export type LocationType = [number, number]
+
+const locationStateOnError = (error: GeolocationError): UnavailableLocationState => {
+  if (error.code === error.PERMISSION_DENIED) {
+    return {
+      status: 'unavailable',
+      message: 'noPermission'
+    }
+  } else if (error.code === error.POSITION_UNAVAILABLE) {
+    return {
+      status: 'unavailable',
+      message: 'notAvailable'
+    }
+  } else {
+    return {
+      status: 'unavailable',
+      message: 'timeout'
+    }
+  }
+}
+
+export type LocationInformationType = {
+  location: LocationType | null
+  locationState: LocationStateType
+  requestAndDetermineLocation: () => Promise<void>
+}
+
+const useLocation = (useSettingsListener = false): LocationInformationType => {
+  const [locationState, setLocationState] = useState<LocationStateType>({
+    status: 'unavailable',
+    message: 'loading'
+  })
+  const [location, setLocation] = useState<LocationType | null>(null)
+
+  const determineLocation = useCallback(() => {
+    Geolocation.getCurrentPosition(
+      (position: GeolocationResponse) => {
+        setLocation([position.coords.longitude, position.coords.latitude])
+        setLocationState({ status: 'ready' })
+      },
+      (error: GeolocationError) => {
+        setLocationState(locationStateOnError(error))
+      },
+      {
+        enableHighAccuracy: true,
```

Comment:
its also used in Landing.tsx
I would recommend to move the settings to a shared constant file